ocs_ci/ocs/ui/acm_ui.py

```diff
@@ -103,3 +129,501 @@ def navigate_credentials_page(self):
+        this function which iteratively fills in values from the dictionary parameter
+
+        key_val (dict): keys corresponds to the xpath of text box, value corresponds
+            to the value to be filled in
+
+        """
+        for xpath, value in key_val.items():
+            self.do_send_keys(locator=xpath, text=value)
+
+    def click_platform_and_credentials(self):
+        self.navigate_create_clusters_page()
+        self.do_click(
+            locator=self.acm_page_nav[PLATFORM_XPATH_MAP[self.platform]], timeout=100
+        )
+        self.do_click(
+            locator=self.acm_page_nav["cc_infrastructure_provider_creds_dropdown"]
+        )
+        credential = format_locator(
+            self.acm_page_nav["cc_infrastructure_provider_creds_select_creds"],
+            self.platform_credential_name,
+        )
+        self.do_click(locator=credential)
+
+    @retry(ACMClusterDeployException, tries=3, delay=10, backoff=1)
+    def goto_cluster_details_page(self):
+        self.navigate_clusters_page()
+        locator = format_locator(self.acm_page_nav["cc_table_entry"], self.cluster_name)
+        self.do_click(locator=locator)
+        self.do_click(locator=self.acm_page_nav["cc_cluster_details_page"], timeout=100)
+        self.choose_expanded_mode(True, self.acm_page_nav["cc_details_toggle_icon"])
+
+    def get_deployment_status(self):
+        self.goto_cluster_details_page()
+        if self.acm_cluster_status_failed(timeout=2):
+            self.deployment_status = "failed"
+        elif self.acm_cluster_status_ready(timeout=2):
+            self.deployment_status = "ready"
+        elif self.acm_cluster_status_creating(timeout=2):
+            self.deployment_status = "creating"
+        else:
+            self.deployment_status = "unknown"
+
+        elapsed_time = int(time.time() - self.deployment_start_time)
+        if elapsed_time > self.cluster_deploy_timeout:
+            if self.deployment_status == "creating":
+                self.deployment_status = "failed"
+                self.deployment_failed_reason = "deploy_timeout"
+
+    def wait_for_cluster_create(self):
+
+        # Wait for status creating
+        staus_check_timeout = 300
+        while (
+            not self.acm_cluster_status_ready(staus_check_timeout)
+            and self.cluster_deploy_timeout >= 1
+        ):
+            self.cluster_deploy_timeout -= staus_check_timeout
+            if self.acm_cluster_status_creating():
+                log.info(f"Cluster {self.cluster_name} is in 'Creating' phase")
+            else:
+                self.acm_bailout_if_failed()
+        if self.acm_cluster_status_ready():
+            log.info(
+                f"Cluster create successful, Cluster {self.cluster_name} is in 'Ready' state"
+            )
+
+    def acm_bailout_if_failed(self):
+        if self.acm_cluster_status_failed():
+            raise ACMClusterDeployException("Deployment is in 'FAILED' state")
+        return
```

Comment:
This return can be removed, it doesn't do anything

Reply:
done